### PR TITLE
Update pytorch-lightning version

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Update PyTorch Lightning version to >=1.6
## Description of the changes
PyTorch Lightning version for `optuna-example` is updated to >=1.6 as [PR #4322](https://github.com/optuna/optuna/pull/4322) is merged into the `master` of `optuna`. Note that DDP is temporally unsupported until [PR #4384](https://github.com/optuna/optuna/pull/4384) is merged into the `master`.
Type annotation is also updated based on the `setup.cfg` in `optuna`.

### CODE REVIEW

File: pytorch/pytorch_lightning_simple.py [hunk 1]
```diff
@@ -53,27 +53,27 @@ def __init__(self, dropout: float, output_dims: List[int]):
...
-    def __init__(self, dropout: float, output_dims: List[int]):
+    def __init__(self, dropout: float, output_dims: List[int]) -> None:
         super().__init__()
         self.model = Net(dropout, output_dims)
 
     def forward(self, data: torch.Tensor) -> torch.Tensor:
         return self.model(data.view(-1, 28 * 28))
 
-    def training_step(self, batch, batch_idx: int) -> torch.Tensor:
+    def training_step(self, batch: torch.Tensor, batch_idx: int) -> torch.Tensor:
```
Comment (nzw0301) [inline pytorch/pytorch_lightning_simple.py:71]: nit 
```suggestion
    def training_step(self, batch: List[torch.Tensor], batch_idx: int) -> torch.Tensor:
```

File: pytorch/pytorch_lightning_simple.py [hunk 2]
```diff
@@ -53,27 +53,27 @@ def __init__(self, dropout: float, output_dims: List[int]):
...
         data, target = batch
         output = self(data)
         return F.nll_loss(output, target)
 
-    def validation_step(self, batch, batch_idx: int) -> None:
+    def validation_step(self, batch: torch.Tensor, batch_idx: int) -> None:
```
Comment (nzw0301) [inline pytorch/pytorch_lightning_simple.py:76]: ditto
```suggestion
    def validation_step(self, batch: List[torch.Tensor], batch_idx: int) -> None:
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]

File: pytorch/pytorch_lightning_simple.py (hunk 1 above)
Comment (Alnusjaponica) [reply pytorch/pytorch_lightning_simple.py]: Thank you for your indication. I applied the changes.

**Review by nzw0301 — APPROVED** [review]
Thanks!

**Review by c-bata — APPROVED** [review]

**Pull request merged**